code/pizzabot.js

```diff
@@ -4,4 +4,73 @@ const pepperoni = "Pepperoni Pizza"
 
 const pizzaPrice = 80
 
-//Put your Javscript code here:
+
+// validateOrderName() which should take the orderName variable as an argument and return true or false if the pizza is on the menu or not.
+// const validateOrderName = (orderName) => {
+//   if ((orderName === hawaiian) || (orderName === vegetarian) || (orderName === pepperoni)) {
+//     return true;
+//   } else {
+//     return false;
+//   }
+// }
+
+// calculateTotalCost() which takes orderQuantity and pizzaPrice as arguments and returns the total cost for the order.
+const calculationTotalCost = (orderQuantity, pizzaPrice) => {
+  return orderQuantity * pizzaPrice;
+}
+
+// calculateCookingTime() which takes orderQuantity and returns the number of minutes it will take to finish the order.
+const calculateCookingTime = (orderQuantity) => {
+  if (orderQuantity <= 2) {
+    return 10;
+  } else if (orderQuantity > 2 && orderQuantity <= 5) {
+    return 15;
+  } else {
+    return 20;
+  }
+}
+
+const greeting = (`Hey! Happy to serve your pizza. On our menu we have ${vegetarian}, ${hawaiian} and ${pepperoni}.`);
+document.getElementById('greeting').innerHTML = greeting;
+
+const greet = (event) => {
+  event.preventDefault();
+  const enterName = ('What is your name?');
+  document.getElementById('enterName-container').style.display = "flex";
+  document.getElementById('enterName').innerHTML = enterName;
+  document.getElementById('nameForm-container').style.display = "flex";
+  document.getElementById('nameForm').style.display = "flex";
+}
+
+const submitName = (event) => {
+  event.preventDefault();
+  const name = document.getElementById('name').value;
+  const orderName = (`${name}, enter the name of the pizza you want to order today.`);
```

Comment:
Hmm, that value could be stored in global variable - there is a lot of code repetition.
Same goes for lines below in each function - as you can see, you are doing same set of things many times, for each step. Couldn't you universalised you code flow so there would be only one function for each step?

Reply:
I created some global variables but, right now, I don't really see how to minimize the code flow in order to keep only one function for each step...

Reply:
You could for example create function :

```
const addFlexStyle  = elementId => {
    document.getElementById(elementId).style.display = "flex";
}

addFlexStyle('pizzaForm-container');
```

Something like that :)

Nevertheless, project is officially approved!